main/api/dependencies/category.py

```diff
@@ -0,0 +1,17 @@
+from fastapi import Depends, status
+from fastapi.exceptions import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from main.api.dependencies.database import get_database_session
+from main.models.category import CategoryModel
+from main.services import category as category_service
+
+
+async def get_category_by_id(category_id: int, session: AsyncSession = Depends(get_database_session)) -> CategoryModel:
+    category = await category_service.get_category_by_id(session, category_id)
+    if category is None:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Cannot find the specified category",
+        )
```

Comment:
Should this be not found exception? Additionally, create a reusable not found exception.